File: ecommerce-js/src/carrito.js
class Carrito {
    //Constructor
    constructor() {
        this.productos = [];
    }

    agregarProducto(producto) {
        this.productos.push(producto);
    }

    calcularTotal() {
        return this.productos.reduce((total, producto) => total + producto.precio, 0);
    }

    aplicarDescuento(porcentaje) {
        return this.calcularTotal() * (1- porcentaje/100);
    }

    estaVacio() {
        return this.productos.length === 0;
    }
}

export default Carrito;

File: ecommerce-java/ecommerce/src/main/java/com/generation/Carrito.java
package com.generation;

import java.util.ArrayList;
import java.util.List;

public class Carrito {
    //Atributo
    private List<Producto> productos = new ArrayList<>();

    //Declaramos los métodos
    public void agregarProducto(Producto producto) {
        this.productos.add(producto);
    }

    public double calcularTotal() {
        return this.productos.stream().mapToDouble(Producto::getPrecio).sum();
    }

    public double aplicarDescuento(double porcentaje) {
        return this.calcularTotal() * (1 - porcentaje/100);
    }

    public boolean estaVacio() {
        return this.productos.isEmpty();
    }

    
    
}
